### ISSUE TITLE
[Scan performance improvements] Define path of site as parameter(s)

### ISSUE THREAD

**Issue by M4Mahdi** [hahwul/dalfox#200]
There is no option for define path of site as parameter(s).
for example:
example.com/search/SearchedTerm{i need to be a param}
or
example.com/video/uid/1234{i need to be a param}/vid/5{i need to be a param}

**Comment by hahwul** [comment]
Hi @M4Mahdi 
Thank you for your opinion. It's a content that I fully sympathize with and I'll develop it quickly so that the function can be applied 🚀

**Comment by hahwul** [comment]
I'm working on it. I think it'll be done soon.

![1414](https://user-images.githubusercontent.com/13212227/110188967-f65a3800-7e60-11eb-8cc3-269dd44def3c.jpg)

**Comment by hahwul** [comment]
## Vuln server for path xss test
https://gist.github.com/hahwul/b648e6d1c8c4a1ac79041024047759a2
```go
package main

import (
    "net/http"
    "time"
    "net/url"

    "github.com/labstack/echo"
    "github.com/tylerb/graceful"
)

func main() {
	e := echo.New()
	e.GET("/", func(c echo.Context) error {
		return c.String(http.StatusOK, "path base xss , vuln server :D")
	})

	// Router & Handler
	e.GET("/users/:name/1234", func(c echo.Context) error {
		name,_ := url.QueryUnescape(c.Param("name"))
		return c.HTML(http.StatusOK, name)
	})
	e.Server.Addr = ":8070"

	// Serve it like a boss
	graceful.ListenAndServe(e.Server, 5*time.Second)
}
```

**Comment by hahwul** [comment]
## Test
![1414](https://user-images.githubusercontent.com/13212227/110226962-b1083a00-7f36-11eb-9d8b-380e037db053.jpg)

**Comment by M4Mahdi** [comment]
> 
> ## Test
> 
> ![1414](https://user-images.githubusercontent.com/13212227/110226962-b1083a00-7f36-11eb-9d8b-380e037db053.jpg)

Thank you!
Is there a manual parameter definition?

**Comment by hahwul** [comment]
@M4Mahdi 
Did you ask about the `-p` option? Please tell me if I misunderstood.
```
  -p, --param string                Only testing selected parameters
```

// and this commit.
If there is no problem after a few tests, it will be released as `v2.3.4`. (Maybe within six hours)

**Comment by hahwul** [comment]
## Testing
- [x] inHTML
```
[*] 🦊 Start scan [SID:Single] / URL: http://localhost:8070/xss/abcd/1
[I] Found 0 testing point in DOM Mining
[I] Content-Type is text/html; charset=UTF-8
[I] Reflected PATH '/xss/dalfoxpathtest/1' => Injected: /inHTML-none(1)]
[V] Triggered XSS Payload (found DOM Object): =<dalfox class=dalfox>
    1 line:  abcd<dalfox class=dalfox>
[POC][V][GET] http://localhost:8070/xss/abcd%3Cdalfox%20class=dalfox%3E/1?=

```
- [x] inJS
```
[I] Reflected PATH '/xss/dalfoxpathtest/2' => Injected: /inJS-single(1)]
[W] Reflected Payload in JS: ='-confirm(1)-'ies ] 
    2 line:    a = 'abcd'-confirm(1)-'' 
[POC][R][GET] http://localhost:8070/xss/abcd'-confirm(1)-'/2?=
[W] Reflected Payload in JS: ='-{{-function(){this.alert(1)}()}}-'
    2 line:    a = 'abcd'-{{-function(){this.alert(1)}()}}-'' 
[POC][R][GET] http://localhost:8070/xss/abcd%27-%7B%7B-function%28%29%7Bthis.alert%281%29%7D%28%29%7D%7D-%27/2?=
[W] Reflected Payload in JS: ='-alert(1)-'
    2 line:    a = 'abcd'-alert(1)-'' 
[POC][R][GET] http://localhost:8070/xss/abcd'-alert(1)-'/2?=
[W] Reflected Payload in JS: ='-{{toString().constructor.constructor('alert(1)')()}}-'
    2 line:    a = 'abcd'-{{toString().constructor.constructor('alert(1)')()}}-'' 
[POC][R][GET] http://localhost:8070/xss/abcd%27-%7B%7BtoString%28%29.constructor.constructor%28%27alert%281%29%27%29%28%29%7D%7D-%27/2?=
[W] Reflected Payload in JS: ='-prompt(1)-'
    2 line:    a = 'abcd'-prompt(1)-'' 
[POC][R][GET] http://localhost:8070/xss/abcd'-prompt(1)-'/2?=
[*] Finish Scan
```
- [x] inATTR
```
[*] 🦊 Start scan [SID:Single] / URL: http://localhost:8070/xss/abcd/3
[I] Found 0 testing point in DOM Mining
[I] Content-Type is text/html; charset=UTF-8
[I] Reflected PATH '/xss/dalfoxpathtest/3' => Injected: /inATTR-single(1)]
[W] Reflected Payload in Attribute: ='onmouseleave=confirm(1) class=dalfox
    1 line:  <div class='abcd'onmouseleave=confirm(1) class=dalfox'>abcd</div>
[POC][R][GET] http://localhost:8070/xss/abcd%27onmouseleave=confirm%281%29%20class=dalfox/3?=
[W] Reflected Payload in Attribute: ='onmouseleave=confirm(1) id=dalfox
    1 line:  <div class='abcd'onmouseleave=confirm(1) id=dalfox'>abcd</div>
[POC][R][GET] http://localhost:8070/xss/abcd%27onmouseleave=confirm%281%29%20id=dalfox/3?=
[W] Reflected Payload in Attribute: ='onpointerenter=prompt`1` class=dalfox
    1 line:  <div class='abcd'onpointerenter=prompt`1` class=dalfox'>abcd</div>
[POC][R][GET] http://localhost:8070/xss/abcd%27onpointerenter=prompt%601%60%20class=dalfox/3?=

```

**Comment by hahwul** [comment]
## Improvements inATTR payloads
```
[*] 🦊 Start scan [SID:Single] / URL: http://localhost:8070/xss/abcd/3
[I] Found 0 testing point in DOM Mining
[I] Content-Type is text/html; charset=UTF-8
[I] Reflected PATH '/xss/dalfoxpathtest/3' => Injected: /inATTR-single(1)]
[V] Triggered XSS Payload (found DOM Object): ='onmouseleave=confirm(1) class=dalfox 
    1 line:  <div class='abcd'onmouseleave=confirm(1) class=dalfox '>abcd</div>
[POC][V][GET] http://localhost:8070/xss/abcd%27onmouseleave=confirm%281%29%20class=dalfox%20/3?=
[*] Finish Scan

```

**Comment by hahwul** [comment]
@M4Mahdi 
Released [`v2.3.4`](https://github.com/hahwul/dalfox/releases/tag/v2.3.4) by modifying some improvements and bugs. You can now scan the distribution version with the modified function. 

Please end the issue and leave it as a [discussions](https://github.com/hahwul/dalfox/discussions) if you have any additional questions!

## Update?
from source
```
$ GO111MODULE=on go get -v github.com/hahwul/dalfox/v2
```

hombrew 
```
$ brew upgrade dalfox
```

snapcraft
```
$ sudo snap refresh dalfox
```

docker 
```
$ docker pull hahwul/dalfox:v2.3.4
```
